geoips/schema/plugin_validator.py

```diff
@@ -0,0 +1,52 @@
+"""Implements valdation machinery for YAML-based plugins."""
+
+import os
+import yaml
+from glob import glob
+from importlib.resources import files
+import jsonschema
+
+
+def extend_with_default(validator_class):
+    validate_properties = validator_class.VALIDATORS["properties"]
+
+    def set_defaults(validator, properties, instance, schema):
+        for property, subschema in properties.items():
+            if "default" in subschema:
+                instance.setdefault(property, subschema["default"])
+
+        for error in validate_properties(
+            validator,
+            properties,
+            instance,
+            schema,
+        ):
+            yield error
+
+    return jsonschema.validators.extend(
+        validator_class,
+        {"properties": set_defaults},
+    )
+
+
+DefaultValidatingValidator = extend_with_default(jsonschema.Draft202012Validator)
```

Comment:
Why does jsonschema use Draft202012Validator....?

Reply:
There are a few ways to tell it which validator to use. You can let it guess which one  you want or you can pin it to a specific validator. I opted to pin to a specific validator to avoid bringing in breaking changes by accident. `Draft202012Validator` is the most recent validator currently available.